PLT-7508: Add settings to disable wide mention confirmation

jespino opened the pull request:
The PR mattermost/mattermost-server#7416 must be merged before it.

Links
-------
mattermost/mattermost-server#7335
https://mattermost.atlassian.net/browse/PLT-7508

Checks
----------
- [X] Has UI changes
- [X] Includes text changes and localization file

esethna commented:
Thanks @jespino! We are working on getting test servers up and running on this repo, once that happens we'll be able to test and review this PR. Appreciate the contribution!

jespino commented:
Perfect, thanks!

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: http://i-06ce05a6c1a487537.spinmint.com:8065/pr18

Test Account 1: Email: `test@test.com` | Password: `passwd`

Test Account 2: Email: `test2@test.com` | Password: `passwd`

Instance ID: i-06ce05a6c1a487537

jespino commented:
I have changed the `@ALL` text and rebased onto master. The default value comes from the server, and the default is defined in this server PR https://github.com/mattermost/mattermost-server/pull/7416/files#diff-1e9c3d615e9ebaaaa3669b4c2fd87d00L78. I can use undefined as "true" but I haven't seen this approach in the code for other settings.

mattermod commented:
Spinmint test server destroyed

mattermod commented:
`Setup Test Server` label detected. Spinmint test server created if build succeeds (checks pass and no conflicts with base branch).

mattermod commented:
Spinmint test server created at: http://i-0c585145b049a9282.spinmint.com:8065/pr18

Test Account 1: Email: `test@test.com` | Password: `passwd`

Test Account 2: Email: `test2@test.com` | Password: `passwd`

Instance ID: i-0c585145b049a9282

mattermod commented:
Spinmint test server destroyed
